sdks/java/harness/src/main/java/org/apache/beam/fn/harness/control/FinalizeBundleHandler.java

```diff
@@ -65,45 +59,13 @@ public static CallbackRegistration create(
     public abstract BundleFinalizer.Callback getCallback();
   }
 
-  private static final Logger LOG = LoggerFactory.getLogger(FinalizeBundleHandler.class);
   private final ConcurrentMap<String, Collection<CallbackRegistration>> bundleFinalizationCallbacks;
   private final PriorityQueue<TimestampedValue<String>> cleanUpQueue;
-  private final Future<Void> cleanUpResult;
 
-  public FinalizeBundleHandler(ExecutorService executorService) {
+  public FinalizeBundleHandler() {
     this.bundleFinalizationCallbacks = new ConcurrentHashMap<>();
     this.cleanUpQueue =
         new PriorityQueue<>(11, Comparator.comparing(TimestampedValue::getTimestamp));
-    this.cleanUpResult =
```

Comment:
is this safe to remove, will the bundleFinalizationCallbacks leak without this task?

Reply:
I'm not sure about this, Is there a way we can test this? Or maybe we can suppress this warning to avoid unexpected side effects.

Reply:
FinalizeBundleHandler is created once per SDK harness process and reused throughout the process lifetime so the cleanup should be kept to avoid memory leak, we can suppress the warning.

Reply:
Thanks, I just added a suppress warning.

Reply:
Should we be inspecting `cleanUpResult` at some point to make sure it worked?